### PR TITLE
[HN-253]: feat: ollama integration

### PR DESCRIPTION
**Pull request by TianYao12**
Added ollama models to work with hive agents

### CODE REVIEW

File: hive_agent/chat/chat_manager.py [hunk 1]
```diff
@@ -17,6 +16,9 @@ def __init__(self, llm, user_id: str, session_id: str):
     async def add_message(
         self, db_manager: DatabaseManager, role: str, content: str
     ):
+        if not isinstance(content, str):
```
Comment (ToJen) [inline hive_agent/chat/chat_manager.py:19]: what's the motivation for this? the params take `content: str` so it'll always be string

File: hive_agent/chat/chat_manager.py [hunk 2]
```diff
@@ -52,7 +54,7 @@ async def generate_response(
             assistant_message = "".join([token async for token in response_stream.async_response_gen()])
         else:
             response = await self.llm.achat(last_message.content, chat_history=chat_history)
-            assistant_message = response.message
+            assistant_message = response.response if hasattr(response, 'response') else str(response)
```
Comment (ToJen) [inline hive_agent/chat/chat_manager.py:57]: is this just for ollama responses i.e. will this stay backwards compatible with existing agents?

File: hive_agent/llms/ollama.py [hunk 3]
```diff
@@ -4,5 +4,32 @@
 from hive_agent.llms.llms import LLMs
 
 
-class OllamaLLM(LLMs):
-    pass
+from llama_index.core.settings import Settings
+from llama_index.core.agent import ReActAgentWorker
+
+
+class OllamaLLM:
```
Comment (ToJen) [inline hive_agent/llms/ollama.py:11]: i'm thinking this should still inherit from the `LLMs` class and then you just override the constructor and create your new agent instance

File: hive_agent/tools/agent_db/nl_query.py [hunk 4]
```diff
@@ -31,7 +28,7 @@ def text_2_sql(db_url: str, prompt: str, tables=None):
     sql_database = SQLDatabase(engine, include_tables=tables)
 
     query_engine = NLSQLTableQueryEngine(
-        sql_database=sql_database, tables=["entries"], llm=llm
+        sql_database=sql_database, tables=["entries"], llm=Settings.llm
```
Comment (ToJen) [inline hive_agent/tools/agent_db/nl_query.py:31]: good catch but this should come from the config. or does this handle that automatically?

File: tests/chat/test_chat_manager.py [hunk 5]
```diff
@@ -12,7 +12,7 @@ async def async_response_gen():
         return type("MockResponse", (), {"async_response_gen": async_response_gen})
 
     async def achat(self, content, chat_history=None):
-        return type("MockResponse", (), {"message": "chat response"})
+        return type("MockResponse", (), {"response": "chat response"})
```
Comment (ToJen) [inline tests/chat/test_chat_manager.py:15]: how does this impact existing model responses

**Review by ToJen — CHANGES_REQUESTED** [review]

File: hive_agent/chat/chat_manager.py (hunk 2 above)
Comment (TianYao12) [reply hive_agent/chat/chat_manager.py:57]: I don't think message is a valid attribute, I get an error.
I've ran it with all the llm types (openai, ollama, mistral, anthropic) and it seems to be working fine.

File: hive_agent/llms/ollama.py (hunk 3 above)
Comment (TianYao12) [reply hive_agent/llms/ollama.py:11]: ok sounds good

File: hive_agent/tools/agent_db/nl_query.py (hunk 4 above)
Comment (TianYao12) [reply hive_agent/tools/agent_db/nl_query.py:31]: Isn't settings global?

File: tests/chat/test_chat_manager.py (hunk 5 above)
Comment (TianYao12) [reply tests/chat/test_chat_manager.py:15]: message was not a valid attribute I believe

File: hive_agent/chat/chat_manager.py (hunk 1 above)
Comment (TianYao12) [reply hive_agent/chat/chat_manager.py:19]: I had an error earlier, but now I removed it and it seems to be owrking fine; for some reason earlier content was AgentResponse, not a string

File: hive_agent/llm_settings.py [hunk 6]
```diff
@@ -22,18 +22,18 @@ def init_llm_settings(config):
...
-        logging.info(f"Model selected is openai")
+        logging.info(f"OpenAI model selected")
     elif "claude" in model:
         Settings.llm = Anthropic(model=model, api_key=os.getenv("ANTHROPIC_API_KEY"))
-        logging.info(f"Model selected is Claude")
+        logging.info(f"Claude model selected")
     elif "llama" in model:
-        Settings.llm = Ollama(model=model, api_key=os.getenv("LLAMA_API_KEY"))
-        logging.info(f"Model selected is Llama")
+        Settings.llm = Ollama(model="llama3", request_timeout=240.0)
```
Comment (ToJen) [inline hive_agent/llm_settings.py:30]: i would add the timeout to the config so that they can specify it themselves

**Review by ToJen — CHANGES_REQUESTED** [review]

File: hive_config_example.toml [hunk 7]
```diff
@@ -3,3 +3,6 @@ model = "gpt-4-turbo-preview"
 
 [environment]
 type = "dev"
+
+[timeout]
+ollama = 500
```
Comment (ToJen) [inline hive_config_example.toml:8]: set to `llm`

File: hive_agent/llm_settings.py [hunk 8]
```diff
@@ -22,18 +22,18 @@ def init_llm_settings(config):
...
-        logging.info(f"Model selected is openai")
+        logging.info(f"OpenAI model selected")
     elif "claude" in model:
         Settings.llm = Anthropic(model=model, api_key=os.getenv("ANTHROPIC_API_KEY"))
-        logging.info(f"Model selected is Claude")
+        logging.info(f"Claude model selected")
     elif "llama" in model:
-        Settings.llm = Ollama(model=model, api_key=os.getenv("LLAMA_API_KEY"))
-        logging.info(f"Model selected is Llama")
+        Settings.llm = Ollama(model="llama3", request_timeout=config.get("timeout", "ollama", 240))
```
Comment (ToJen) [inline hive_agent/llm_settings.py:30]: you would have to make a new variable and apply it to all the models, default should be 30

**Review by ToJen — CHANGES_REQUESTED** [review]

**Review by ToJen — APPROVED** [review]

**Pull request merged**